### PR TITLE
Update prometheus related crates

### PR DESCRIPTION
**Pull request by KeneePatel**
Time spent on this PR: 2D

Resolves: #971 

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?
The dependency for prometheus related crates are updated to the latest versions as of writing this PR. Two new dependencies of http-body-util and hyper-util gets added. The implementation of init_prometheus gets slightly updated. 

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/prometheus_handler/mod.rs [hunk 1]
```diff
@@ -87,32 +90,40 @@ pub async fn init_prometheus(prometheus_addr: SocketAddr, registry: Registry) ->
...
+        // Manufacturing a connection with the service from request_metrics fn
+        let conn = server::conn::auto::Builder::new(TokioExecutor::new()).serve_connection(
+            io,
+            service_fn(move |req: Request<hyper::body::Incoming>| request_metrics(req, registry.clone())),
+        );
 
-    let result = server.await.map_err(Into::into);
+        // awaiting for the res
+        // TODO: converting the Box<dyn Error + Send + Sync> to Error. having troubles with that
+        let res = conn.await.map_err(|_| Error::Http(hyper::http::Error::from("Some error occured")));
```
Comment (KeneePatel) [inline src/prometheus_handler/mod.rs:111]: Stuck at this part converting Box<dyn Error + Send + Sync> to Error

File: Cargo.toml [hunk 2]
```diff
@@ -84,13 +84,9 @@ use_self = "allow"
 
 [dependencies]
 # Starknet dependencies
-cainome = { git = "https://github.com/cartridge-gg/cainome.git", tag = "v0.2.5", default-features = false, features = [
-  "abigen-rs",
-] }
+cainome = { git = "https://github.com/cartridge-gg/cainome.git", tag = "v0.2.5", default-features = false, features = ["abig-rs"] }
```
Comment (tcoratger) [inline Cargo.toml:87]: Why the `features` name is modified here?

File: src/prometheus_handler/mod.rs (hunk 1 above)
Comment (tcoratger) [reply src/prometheus_handler/mod.rs:111]: For simplicity I would say that you could do something like


```rust
#[derive(Debug, thiserror::Error)]
pub enum Error {
    /// Hyper internal error.
    #[error(transparent)]
    Hyper(#[from] hyper::Error),

    /// Http request error.
    #[error(transparent)]
    Http(#[from] Box<dyn std::error::Error + Send + Sync>),

    /// i/o error.
    #[error(transparent)]
    Io(#[from] std::io::Error),

    #[error("Prometheus port {0} already in use.")]
    PortInUse(SocketAddr),
}
```

and then here:

```rust
        let res = conn.await.map_err(|e| Error::Http(e.into()))?;
```

as we just want to capture the error in its simplest format here.

File: Cargo.toml (hunk 2 above)
Comment (KeneePatel) [reply Cargo.toml]: yeah this was a typo. Sorry I didn't realise it.

Comment (KeneePatel) [reply Cargo.toml]: Should I run format on the `Cargo.toml` as well for consistency, or make it like it was earlier i.e. some features on new line in extended form, while few others on the same line.

File: src/prometheus_handler/mod.rs (hunk 1 above)
Comment (KeneePatel) [reply src/prometheus_handler/mod.rs]: Yeah, I changed the code accordingly

**Comment by tcoratger** [comment]
Very small nits, pending failing clippy

**Comment by KeneePatel** [comment]
clippy is like the middle child for my brain. I always forget it. anyways, I fixed it. gunna put up a post-it note to fix it before commits.

**Review by tcoratger — DISMISSED** [review]

**Comment by tcoratger** [comment]
lgtm

**Review by tcoratger — APPROVED** [review]

**Pull request merged**